app/src/main/java/org/p2p/wallet/splash/SplashPresenter.kt

```diff
@@ -2,15 +2,22 @@ package org.p2p.wallet.splash
 
 import timber.log.Timber
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.p2p.core.common.di.AppScope
 import org.p2p.wallet.auth.analytics.OnboardingAnalytics
 import org.p2p.wallet.auth.interactor.AuthInteractor
 import org.p2p.wallet.common.mvp.BasePresenter
-import org.p2p.wallet.user.interactor.UserInteractor
+import org.p2p.wallet.home.events.AppLoader
+import org.p2p.wallet.tokenservice.TokenServiceCoordinator
+import org.p2p.wallet.user.interactor.BlockChainTokensMetadataInteractor
 
 class SplashPresenter(
     private val authInteractor: AuthInteractor,
     private val onboardingAnalytics: OnboardingAnalytics,
-    private val userInteractor: UserInteractor
+    private val blockchainTokensInteractor: BlockChainTokensMetadataInteractor,
```

Comment:
```suggestion
    private val blockchainTokensInteractor: BlockchainTokensMetadataInteractor,
```
Blockchain - you mean Solana network?